Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -2051,9 +2065,36 @@ extension SourceKitLSPServer {
     originatorUri: DocumentURI,
     languageService: LanguageService
   ) async throws -> Location {
+    // Check if we're already in the target interface with the same module/group/symbol
+    if case .generatedInterface(let interfaceData) = try? ReferenceDocumentURL(from: originatorUri),
```

Comment:
This comment still applies here: https://github.com/swiftlang/sourcekit-lsp/pull/2233/#discussion_r2267852544